satpy/writers/__init__.py

```diff
@@ -180,17 +180,24 @@ def _determine_mode(dataset):
 
 def add_overlay(orig, area, coast_dir, color=(0, 0, 0), width=0.5, resolution=None,
```

Comment:
How about moving out the decoration functions to a separate module for clarity ?

Reply:
We could and probably should, but I'm not sure these functions are mature enough for this. I'm still not happy with how these have to be used and would like to define a better interface, but I don't have an idea (or the time) to figure out what that would look like.

Reply:
We can keep that for later, maybe along with a renovation of pycoast

Reply:
Yeah good idea. Something more xarray and/or dask friendly. It would be *really* cool if pycoast/pydecorate had a better interface for handling xarray objects. Imagine if you could do `new_data_array = my_data_array.pycoast.draw_coastlines(...)`. Might be a really good feature to advertise to the crowds that are used to working with cartopy.